packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -402,4 +451,170 @@ void main() {
 
     handle.dispose();
   });
+
+  testWidgets('Ensure canTapOnHeader is false by default',  (WidgetTester tester) async {
+    final expansionPanel = ExpansionPanel(
+      headerBuilder: (BuildContext context, bool isExpanded) =>
+        const Text('Demo'),
+      body: const SizedBox(height: 100.0),
+    );
+    expect(expansionPanel.canTapOnHeader, isFalse);
+  });
+
+  testWidgets('Toggle radio panel list when clicking on header',  (WidgetTester tester) async {
+    const Key firstPanelKey = Key('firstPanelKey');
+    const Key secondPanelKey = Key('secondPanelKey');
+
+    final List<ExpansionPanel> _demoItemsRadio = <ExpansionPanelRadio>[
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'B' : 'A', key: firstPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        value: 0,
+        canTapOnHeader: true,
+      ),
+      ExpansionPanelRadio(
+        headerBuilder: (BuildContext context, bool isExpanded) {
+          return Text(isExpanded ? 'D' : 'C', key: secondPanelKey);
+        },
+        body: const SizedBox(height: 100.0),
+        value: 1,
+        canTapOnHeader: true,
+      ),
+    ];
+
+    final ExpansionPanelList _expansionListRadio = ExpansionPanelList.radio(
+      children: _demoItemsRadio,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SingleChildScrollView(
+          child: _expansionListRadio,
+        ),
+      ),
+    );
+
+    // Initializes with all panels closed
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+
+    await tester.tap(find.byKey(firstPanelKey));
+    await tester.pumpAndSettle();
+
+    // Now the first panel is open
+    expect(find.text('A'), findsNothing);
+    expect(find.text('B'), findsOneWidget);
+    expect(find.text('C'), findsOneWidget);
+    expect(find.text('D'), findsNothing);
+
+    await tester.tap(find.byKey(secondPanelKey));
+    await tester.pumpAndSettle();
+
+    // Now the second panel is open
+    expect(find.text('A'), findsOneWidget);
+    expect(find.text('B'), findsNothing);
+    expect(find.text('C'), findsNothing);
+    expect(find.text('D'), findsOneWidget);
+  });
+
+  testWidgets('Toggle simple panel list by clicking on header when canTapOnHeader is true',  (WidgetTester tester) async {
```

Comment:
Test with `ExpansionPanelRadio` as well

Reply:
Do you have something else in mind, other than what I do in 'Toggle radio panel list when clicking on header'?